docs/feature/subnet-scarcity/phase-3/2-nncbeta.md

```diff
@@ -0,0 +1,70 @@
+## Revising the NodeNetworkConfig to v1beta1 [[Phase 3 Design]](../proposal.md#3-2-revise-the-nnc-to-v1beta1)
+
+As some responsibility is shifted out of the NodeNetworkConfig (the Scaler), and the use-cases evolve, the NodeNetworkConfig needs to be updated to remain adaptable to all scenarios. Notably, to support multiple NetworkContainers per Node, the NNC should acknowledge that those may be from separate Subnets, and should map the `requestedIPCount` per known NetworkContainer. With the ClusterSubnet CRD hosting the Subnet Scaler properties, this will allow Subnets to scale independently even when they are used on the same Node.
+
+Since this is a significant breaking change, the NodeNetworkConfig definition must be incremented. Since the spec is being incremented, some additional improvements are included.
+
+```diff
+-   apiVersion: acn.azure.com/v1alpha
++   apiVersion: acn.azure.com/v1beta1
+    kind: NodeNetworkConfig
+    metadata:
+        name: nodename
+        namespace: kube-system
+    spec:
+-       requestedIPCount: 16
+-       ipsNotInUse:
++       releasedIPs:
+        -   abc-ip-123-guid
++       secondaryIPs:
++           abc-nc-123-guid: 16
+    status:
+-       assignedIPCount: 1
+        networkContainers:
+        -   assignmentMode: dynamic
+            defaultGateway: 10.241.0.1
+            id: abc-nc-123-guid
+-           ipAssignments:
+-           -   ip: 10.241.0.2 
+-               name: abc-ip-123-guid
+            nodeIP: 10.240.0.5
+            primaryIP: 10.241.0.38
+            resourceGroupID: rg-id
++           secondaryIPCount: 1
++           secondaryIPs:
++           -   address: 10.241.0.2 
++               id: abc-ip-123-guid
+            subcriptionID: abc-sub-123-guid
+            subnetAddressSpace: 10.241.0.0/16
+            subnetID: podnet
+-           subnetName: podnet
+            type: vnet
+            version: 49
+            vnetID: vnet-id
+-       scaler:
+-           batchSize: 16
+-           maxIPCount: 250
+-           releaseThresholdPercent: 150
+-           requestThresholdPercent: 50
+        status: Updating
+```
+
+In order:
+- the GV is incremented to `acn.azure.com/v1beta1`
+- the `spec.requestedIPCount` key is renamed to `spec.secondaryIPs`
+    - the value is change from a single scalar to a map of `NC ID` to scalar values
+- the `spec.ipsNotInUse` key is renamed to `spec.releasedIPs`
+- the `status.assignedIPCount` field is moved and renamed to `status.networkContainers[].secondaryIPCount`
+- the `status.networkContainers[].ipAssignments` field is renamed to `status.networkContainers[].secondaryIPs`
+    - the keys of the secondaryIPs are renamed from `ip` and `name` to `address` and `id` respectively
+- the `status.subnetName` fields is removed as a duplicate of `status.subnetID`
```

Comment:
subnetName and subnetID are not the same things. SubnetName can be derived from the subnet ID. Subnet ID should ideally hold the GUID associated with the subnet something like /subscription/<sub>/resourceGroups/<rg>/Microsoft.Network/Vnet/<vnetname>/subnet/<subnetname>

Reply:
subnetID is not the ARM ID, subnetName and subnetID are the same and redundant

Reply:
updated to drop subnetID and keep subnetName as the value we are using is actually the name.